### PR TITLE
FlightPlanner: Added support for Itinerary.layoverTimes

### PR DESCRIPTION
**Pull request by meghanarayanan**

### CODE REVIEW

File: src/main/scala/training/scala/air_scala/flights/Flight.scala [hunk 1]
```diff
@@ -101,15 +101,9 @@ object FlightPlanner {
...
+  def filterByMinConnectionTime(minConnectionTime: Duration)(itineraries: Set[ProposedItinerary]): Set[ProposedItinerary] =
     itineraries.filter { it =>
-      it.flights.sliding(2).forall {
-        case Seq(f1, f2) =>
-          val connectionTime = Duration.millis(f2.schedule.origin.time.getMillis - f1.schedule.destination.time.getMillis)
-          connectionTime >= minConnectionTime
-        case _ => true
-      }
-    }
+    val layovers: Seq[_root_.com.github.nscala_time.time.Imports.Duration] = Itinerary.layoverTimes(it)
```
Comment (jgagnon1) [inline src/main/scala/training/scala/air_scala/flights/Flight.scala]: We should try to not used FQN as it descreases readbility - couldn't you use the imported one ?

File: src/main/scala/training/scala/air_scala/flights/scheduling/Scheduling.scala [hunk 2]
```diff
@@ -8,7 +8,15 @@ import training.scala.air_scala.flights.Flight
 import scala.annotation.tailrec
 
 
-object Itinerary { }
+object Itinerary {
+  def layoverTimes(itinerary: ProposedItinerary): Seq[Duration] = itinerary.flights match {
```
Comment (jgagnon1) [inline src/main/scala/training/scala/air_scala/flights/scheduling/Scheduling.scala]: I don't think you need the first case as it'll be catched in the sliding map

File: src/main/scala/training/scala/air_scala/flights/Flight.scala (hunk 1 above)
Comment (meghanarayanan) [inline src/main/scala/training/scala/air_scala/flights/Flight.scala]: Will remove the type check, I added it for debugging through IntelliJ, and forgot to remove it

**Comment by jgagnon1** [comment]
+1 thanks!

**Pull request merged**